doc/spec/PluginAPI.md

```diff
@@ -0,0 +1,16 @@
+# Sia-UI Plugin API Specification
+
+## Introduction
+
+This specification outlines the functionality that Sia-UI's plugin API exposes to developers.
+
+## Desired Functionality
+
+The plugin API should provide an importable javascript module that provides all of the following features:
+
+- A function which registers a plugin with Sia-UI.
+- Access to the running, configured, `sia.js` wrapper.
+- Functions to open and save files on the host.
+- Functions to control the appearance of the plugin.  This includes the ability to set the icon for the plugin and the plugin's order in the sidebar.
```

Comment:
What's your opinion on providing a function for setting the icon vs. loading it automatically from the plugin folder? My thoughts:
- Function: more flexible than requiring an icon in a specific place; can be (ab)used to make animated icons or notification icons; might be forgotten by plugin author
- File: harder to forget; reduces API surface; not programmable; increases coupling

Also, I don't think plugins should have control over their ordering in the sidebar. That seems best left to the user.